dnf5/commands/check/check.cpp

```diff
@@ -0,0 +1,336 @@
+// It is initialized with values from the configuration.
+std::vector<std::string> installonly_pkgs_provides;
+
+// Contains a set of problems for each problematic package.
+// The std::map Key is the "nevra" of the problematic package.
+std::map<std::string, std::set<Problem>> problems;
+
+// Contains a list of installed "nevras" for each "name.arch".
+// For "installonly" packages, only one "nevra" is present.
+// Usually contains 1 "nevra" for "name.arch". Multiple "nevras" means "duplicates".
+std::map<std::string, std::vector<std::string>> installed_na_nevras;
+
+
+cli::ArgumentParser::NamedArg * create_option(
+    cli::ArgumentParser & parser, const std::string & name, const std::string & descr, bool & opt_to_modify) {
+    auto option = parser.add_new_named_arg(name);
+    option->set_long_name(name);
+    option->set_description(descr);
+    option->set_parse_hook_func([&opt_to_modify](cli::ArgumentParser::NamedArg *, const char *, const char *) {
+        opt_to_modify = true;
+        return true;
+    });
+    return option;
+}
+
+
+inline bool is_package_installonly(Header h) noexcept {
+    rpmtd td = rpmtdNew();
+    headerGet(h, RPMTAG_PROVIDES, td, 0);
+    while (auto provide = rpmtdNextString(td)) {
+        for (const auto & installonly_pkg_provide : installonly_pkgs_provides) {
+            if (installonly_pkg_provide == provide) {
+                rpmtdFree(td);
+                return true;
+            }
+        }
+    }
+    rpmtdFree(td);
+
+    return false;
+}
+
+
+// Ensures that NEVRA always contains an EPOCH.
+// Returns the input `nevra`. EPOCH is inserted if it was not on the input.
+// librpm functions such as "rpmProblemGetPkgNEVR()" and "rpmProblemGetAltNEVR()" return epoch 0 only sometimes.
+// We want the epoch always.
+inline std::string ensure_full_nevra(const char * nevra) {
+    auto nevras = rpm::Nevra::parse(nevra, {rpm::Nevra::Form::NEVRA});
+    return rpm::to_full_nevra_string(nevras.front());
+}
+
+
+// Finds problems for the installed package defined by Header h.
+// Principle: Adds the installed package to an empty transaction and checks for errors.
+int find_problems(QVA_t qva, rpmts ts, Header h) {
+    int rc = 0;
+
+    if (qva->qva_flags & VERIFY_DEPS) {
+        rpmtsEmpty(ts);
+        (void)rpmtsAddInstallElement(ts, h, NULL, 0, NULL);
+
+        (void)rpmtsCheck(ts);
+        auto te = rpmtsElement(ts, 0);
+        auto ps = rpmteProblems(te);
+        rc = rpmpsNumProblems(ps);
+
+        if (rc > 0) {
+            rpmpsi psi = rpmpsInitIterator(ps);
+            while (auto p = rpmpsiNext(psi)) {
+                auto problem_cstr = rpmProblemGetStr(p);
+                libdnf_assert(problem_cstr != NULL, "command \"check\": rpmProblemGetStr() returns NULL");
+                auto cnevr = rpmProblemGetPkgNEVR(p);
+                libdnf_assert(cnevr != NULL, "command \"check\": rpmProblemGetPkgNEVR() returns NULL");
+                auto alt_cnevr = rpmProblemGetAltNEVR(p);
```

Comment:
If `rpmProblemGetAltNEVR(p)` returns `nevra` formatted string, may I ask you to rename the variable according the format of the string?

Reply:
Renamed: `cnevr` -> `cnevra`, `alt_cnevr` -> `alt_cnevra`, `nevr` -> `nevra`, `alt_nevr` -> `alt_nevra`
Added `const` to more places.